clients/fides-js/src/lib/consent.tsx

```diff
@@ -0,0 +1,305 @@
+    transform: translateY(100%);
+}
+
+div#fides-consent-banner.fides-consent-banner-top {
+    position: fixed;
+    z-index: 1;
+    width: 100%;
+    top: 0;
+    left: 0;
+    transform: translateY(0%);
+}
+
+div#fides-consent-banner.fides-consent-banner-top.fides-consent-banner-hidden {
+    transform: translateY(-100%);
+}
+
+
+div#fides-consent-banner-description {
+    margin-top: 0.5em;
+    margin-right: 2em;
+    min-width: 33%;
+    flex: 1;
+}
+
+div#fides-consent-banner-buttons {
+    display: flex;
+    flex-direction: row;
+    flex-wrap: wrap;
+}
+
+button.fides-consent-banner-button {
+    display: inline-block;
+    flex: auto;
+    cursor: pointer;
+    text-align: center;
+    margin: 0;
+    margin-top: 0.25em;
+    margin-right: 0.5em;
+    padding: var(--fides-consent-banner-button-padding);
+    background: var(--fides-consent-banner-button-background);
+    color: var(--fides-consent-banner-button-text-color);
+    border: none;
+    border-radius: var(--fides-consent-banner-button-border-radius);
+
+    font-family: inherit;
+    font-size: 100%;
+    line-height: 1.15;
+    text-decoration: none;
+}
+
+button.fides-consent-banner-button:hover {
+    background: var(--fides-consent-banner-button-hover-background);
+}
+
+button.fides-consent-banner-button.fides-consent-banner-button-primary {
+    background: var(--fides-consent-banner-button-primary-background);
+    color: var(--fides-consent-banner-button-primary-text-color);
+}
+
+button.fides-consent-banner-button.fides-consent-banner-button-primary:hover {
+    background: var(--fides-consent-banner-button-primary-hover-background);
+}
+
+button.fides-consent-banner-button.fides-consent-banner-button-secondary {
+    background: var(--fides-consent-banner-button-secondary-background);
+    color: var(--fides-consent-banner-button-secondary-text-color);
+}
+
+button.fides-consent-banner-button.fides-consent-banner-button-secondary:hover {
+    background: var(--fides-consent-banner-button-secondary-hover-background);
+}
+
+button.fides-consent-banner-button.fides-consent-banner-button-tertiary {
+    background: var(--fides-consent-banner-button-tertiary-background);
+    color: var(--fides-consent-banner-button-tertiary-text-color);
+}
+
+button.fides-consent-banner-button.fides-consent-banner-button-tertiary:hover {
+    background: var(--fides-consent-banner-button-tertiary-hover-background);
+}`
+
+/**
+ * Builds a 'style' element containing the CSS styles for the consent banner
+ */
+function BuildStyles() {
+    return (<style>
+        {defaultStyle}
+    </style>)
+}
+
+/**
+ * Initialize the Fides Consent Banner or Link, with optional extraOptions to override defaults.
+ *
+ * (see the type definition of ConsentBannerOptions for what options are available)
+ */
+export const InitFidesConsent = async(defaults: CookieKeyConsent, extraOptions?: ConsentBannerOptions): Promise<void> => {
```

Comment:
I *really* dislike that the consent `defaults` are forced to be passed into this. This was a messy part of my POC so I do want to untangle that- it doesn't provide good separation of concerns for the banner code itself to be aware of the consent preferences configs, etc.